File: pkg/envx/env.go
// Package envx provides the shared optional-env readers (Go counterpart of
// the classifier's Env.java). Required variables stay explicit in each
// service's validate — only defaulted reads live here.
package envx

import (
	"os"
	"strconv"
)

// Int reads an env var as a positive int, returning fallback when unset,
// unparsable, or < 1.
func Int(name string, fallback int) int {
	raw := os.Getenv(name)
	if raw == "" {
		return fallback
	}
	value, err := strconv.Atoi(raw)
	if err != nil || value < 1 {
		return fallback
	}
	return value
}

File: services/classifier-job/src/main/java/io/datacat/classifier/config/Env.java
package io.datacat.classifier.config;

/** Generic env readers with fallbacks. */
public final class Env {

	private Env() {
	}

	public static String string(String name, String fallback) {
		String raw = System.getenv(name);

		return raw == null || raw.isBlank() ? fallback : raw;
	}

	public static long longValue(String name, long fallback) {
		String raw = System.getenv(name);

		return raw == null || raw.isBlank() ? fallback : Long.parseLong(raw);
	}
}
